### PR TITLE
[BEAM-1909] BigQuery read transform fails for DirectRunner when querying non-US regions

### PR DESCRIPTION
**Pull request by ubunatic**
**Note**: This is an updated and condensed PR for the original PR: https://github.com/apache/beam/pull/2509

**Description**:
I partially (!) fixed the issue by getting the location of the source table. Then I use this location as location of the created temp dataset. The added parameters are optional and should not break anything.

**Tests**: I tested with an US and EU dataset as source in a `DirectRunner`, i.e., using a `BigQuerySource` with `table=<some-table>`.

**What's missing**: It does not work for a `BigQuerySource` with `query=<some-query>`.

The corresponding Jira issue should not yet be closed.

### CODE REVIEW

**Comment by ubunatic** [comment]
@sb2nov @chamikaramj please use this updated PR

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 1]
```diff
@@ -607,7 +607,20 @@ def __init__(self, source, test_bigquery_client=None, use_legacy_sql=True,
 
   def __enter__(self):
     self.client = BigQueryWrapper(client=self.test_bigquery_client)
-    self.client.create_temporary_dataset(self.executing_project)
+
+    tr = self.source.table_reference
```
Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Can we move this whole section inside the get_table_location function?
`def _get_source_table_location(self)`

and this `__enter__` would just look like `self.client.create_temporary_dataset(self.executing_project, self._get_source_table_location())`

**Comment by sb2nov** [comment]
Thanks, just one comment otherwise looks great.

File: sdks/python/apache_beam/io/gcp/bigquery.py (hunk 1 above)
Comment (ubunatic) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Done.

File: sdks/python/apache_beam/io/gcp/bigquery.py [hunk 2]
```diff
@@ -867,7 +883,15 @@ def _delete_dataset(self, project_id, dataset_id, delete_contents=True):
   @retry.with_exponential_backoff(
       num_retries=MAX_RETRIES,
       retry_filter=retry.retry_on_server_errors_and_timeout_filter)
-  def create_temporary_dataset(self, project_id):
+  def get_table_location(self, project_id, dataset_id, table_id):
```
Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: do we need this function anymore?

**Comment by sb2nov** [comment]
R: @chamikaramj PTAL

**Comment by chamikaramj** [comment]
Please fix the test failures.

======================================================================
ERROR: test_read_from_query (apache_beam.io.gcp.bigquery_test.TestBigQueryReader)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_MavenInstall/sdks/python/apache_beam/io/gcp/bigquery_test.py", line 433, in test_read_from_query
    with beam.io.BigQuerySource(query='query').reader(client) as reader:
  File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_MavenInstall/sdks/python/apache_beam/io/gcp/bigquery.py", line 624, in __enter__
    self.executing_project, location=self._get_source_table_location())
  File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_MavenInstall/sdks/python/apache_beam/io/gcp/bigquery.py", line 610, in _get_source_table_location
    if tr.projectId is None:
AttributeError: 'NoneType' object has no attribute 'projectId'

File: sdks/python/apache_beam/io/gcp/bigquery.py (hunk 2 above)
Comment (ubunatic) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: yes, it is used in line 617:

    source_location = self.client.get_table_location(
            source_project_id, source_dataset_id, source_table_id)

alternatively, we add a `client.get_table` function and get the `table.location` from there. There is already a `client.get_or_create_table` function, but I would refrain from using for getting only the `location`.

Comment (sb2nov) [inline sdks/python/apache_beam/io/gcp/bigquery.py]: Got it. Looks good

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Pull request closed without merging**